core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -678,5 +680,14 @@ public void createNewByteArrayInputStream() throws Exception {
     assertNotNull( stream );
   }
 
+  @Test
+  public void extractXmiFromZipTest() throws Exception {
+    File f = new File( "target/test-classes/import_metadata.zip" );
+    FileInputStream metadataFile = new FileInputStream( f );
+    FormDataContentDisposition schemaFileInfo = mock( FormDataContentDisposition.class );
+    when( schemaFileInfo.getFileName() ).thenReturn( f.getName() );
+    InputStream is = metadataService.extractXmiFile(  metadataFile, schemaFileInfo );
+    assertNotNull( is );
```

Comment:
Shouldn't you test that you get the expected xmi file?
The function could return a temp file or an empty file and the test would still pass.